### PR TITLE
[Replace Non-Blocking Usage] LPS-197550 MainServlet

### PR DESCRIPTION
**Pull request by haydenzheng314**
Ticket: https://liferay.atlassian.net/browse/LPS-197550
cc @dantewang 
Thanks

### CODE REVIEW

**Comment by LisztLiszt** [comment]
ci:test:relevant

**Comment by LisztLiszt** [comment]
ci:test:startup

File: portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java [hunk 1]
```diff
@@ -134,6 +134,8 @@ public class MainServlet extends HttpServlet {
 	public void destroy() {
 		ShutdownHelperUtil.setShutdown(true);
 
+		_serviceTracker.close();
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java:137]: @LisztLiszt No need to close a static one.

File: portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java [hunk 2]
```diff
@@ -1325,12 +1329,20 @@ ServletContext.class, getServletContext(),
 
 	private static final Log _log = LogFactoryUtil.getLog(MainServlet.class);
 
+	private static final BundleContext _bundleContext =
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java:1332]: @LisztLiszt Please inline this field.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by LisztLiszt** [comment]
@dantewang Force pushed

**Comment by zhangyan0701** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java [hunk 3]
```diff
@@ -1328,9 +1328,18 @@ ServletContext.class, getServletContext(),
...
+
+	static {
+		ServiceTracker<ReleaseManager, ReleaseManager> serviceTracker =
+			new ServiceTracker<>(
+				SystemBundleUtil.getBundleContext(), ReleaseManager.class,
+				null);
+
+		serviceTracker.open();
+
+		_releaseManager = serviceTracker.getService();
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java:1341]: @LisztLiszt This is more than what I commented. Actually, please keep getting the service when it's used. Only keep the service tracker a static field, open it in static block.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by LisztLiszt** [comment]
@dantewang Force pushed, only initialize and open the `_serviceTracker` in static block causes SF [failure](https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/static_block_check.markdown)

**Comment by LisztLiszt** [comment]
ci:test:relevant

**Comment by LisztLiszt** [comment]
ci:test:startup

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
ci:test:upgrade

**Comment by dantewang** [comment]
@LisztLiszt Please check CI results, you can refer to the `[Resources for CI test and PR analysis]` in core-infra private channel.

**Comment by LisztLiszt** [comment]
@dantewang Checked with Summer, no regression found

**Comment by dantewang** [comment]
ci:forward:force

cc @shuyangzhou After this one there will be no real usage of ServiceProxyFactory.

**Comment by dantewang** [comment]
ci:forward:force

**Pull request closed without merging**